### ISSUE TITLE
Considerations for progressive enhancement

### ISSUE THREAD

**Issue by theetrain** [carbon-design-system/carbon-components-svelte#1622]
The following items are needed in order to enable progressive enhancement capabilities for prerendered or server-rendered components:

## General: Form actions and `href`

See #1599 

- [ ] components with an inner button should provide a `[target]FormActions` prop that receives a shape of
`{ type?: 'GET' | 'POST'; action?: String; enhance?: fn }`
Where `enhance` is an implied SvelteKit Form enhance action, but can be any Svelte action
And `action` is the target url for `<form action={action} />`
- [ ] components with inner buttons that can change data or state may receive an `href` prop, allowing users to choose a new page or the same page with `searchParams`, impacting the displayed information

## UIShell

WIP
- [ ] use media queries for responsive styles

## Accordion

WIP

## Breakpoint

- [ ] discourage use for progressive enhancement
- [ ] Consider providing utility classes or components for responsive show/hide logic

## `<Button>`

- [ ] `<Button>` uses `$$restProps`, so it should be possible to forward the HTML `formaction` attribute. 
- [ ] it's easy enough to wrap `<Button />` with a plain `<form />`, so nothing further is required

## Modal

WIP

## Combobox

WIP

## Dropdown

WIP

## Tabs

WIP

## Toggle

WIP

## Tooltip, Tooltip Icon, Tooltip Definition

WIP

**Comment by gohde** [comment]
In terms of using sveltekit's form actions, would it be possible to add an option to the modal to have the primary button be of type submit? or have the buttons as a slot in order to pass in custom buttons. That way we can wrap the modal in a native form and have the whole thing work as intended by sveltekit?

**Comment by theetrain** [comment]
@gohde absolutely. Though the actual implementation is yet-decided, we'll try to find a way to make the modal's buttons submit as a form. The tricky thing with some modals is they don't open easily without JS, so part of the exploration is to find a pattern that would allow the page to fully refresh without JS and display an opened modal that has a submittable form.

**Comment by gohde** [comment]
I had a similar issue with Tabs and ended up solving it using route params. As you tabbed around it would update the url. It worked fine and I don't see why that couldn't work with this also. [[modal]] or with nested or several modals using [...modals]?

**Comment by theetrain** [comment]
I'm sure it's achievable, I guess I was needlessly concerned with a potential refactor for Modal, but we'll cross that bridge in time.

Aspirations include:

- Tabs will support an `href` prop, and `<button formaction="" />`
- Modals can be opened with CSS when the `open` prop is `true` in  the case of a page reload; whether navigating to a new url or via query params such as `/page?confirm=true` that yields an SSR-opened modal. Modal buttons can support pass-through props such as:

   ```html
   <Modal primaryButton={{ form: "confirm-form", href }}>
     <form id="confirm-form">
     </form>
   </Modal>
   ```

**Comment by brunnerh** [comment]
Noticed a few issues with the current `Tabs`:
- The `selected` highlighting will not apply without JS.
  There is some complex context dance between `Tabs` and `Tab` and the `selected` state in `Tab` is not updated in time for SSR.
  I have not found a simple workaround for this yet.
- Making the links on the tabs work as expected requires some additional logic because the clicks are intercepted with JS.
  (#2165)
  For SvelteKit, I added `goto` calls to synchronize the client-side router:
  ```ts
  function onClickTab(e: MouseEvent & { currentTarget: HTMLLIElement }) {
  	const target = e.currentTarget.querySelector('a')!;
  	goto(target.href);
  }
  ```
  ```svelte
  <Tab label="A" href="/page/tab-a" onclick={onClickTab} />
  <Tab label="B" href="/page/tab-b" onclick={onClickTab} />
  ```
- `Tabs` expects one `TabContent` per specified tab, but without JS, all contents will be visible by default.
  Using something like `{#if tab == 'a'}` around the `TabContent` components will break the component when JS is loaded.
  As a workaround, I moved the `{#if}` into the `TabContent`. This has one more wrinkle in that this component has some padding which will still be visible in the non-JS case. One can override the styles and move those also into the conditionally shown contents.
  ```svelte
  <svelte:fragment slot="content">
  	<TabContent class="p0">
  		{#if tab == 'a'}
  			<div class="p16">...</div>
  		{/if}
  	</TabContent>
  	<TabContent class="p0">
  		{#if tab == 'b'}
  			<div class="p16">...</div>
  		{/if}
  	</TabContent>
  </svelte:fragment>
  ```
  (There could be a simple CSS solution to this when [`:empty` matches whitespace as well](https://caniuse.com/mdn-css_selectors_empty_matches_whitespace), to just hide any empty tab content elements.)
